src/serializer/serializer.js

```diff
@@ -106,6 +106,7 @@ export class Serializer {
     if (this.options.additionalFunctions) {
       this.functions.checkThatFunctionsAreIndependent();
     }
+    // TODO: think about how this interacts with serializing extra functions
```

Comment:
The `initializeMoreModules` code should run *before* we collect effects on additional functions. So the code above needs to be moved below. Basically, the additional functions must come last.